packages/salesforcedx-vscode-core/src/orgBrowser/orgMetadata.ts

```diff
@@ -0,0 +1,157 @@
+      cwd: getRootWorkspacePath()
+    }).execute(cancellationToken);
+
+    execution.processExitSubject.subscribe(async data => {
+      this.logMetric(execution.command.logName, startTime);
+      buildTypesList(this.outputPath);
+    });
+    notificationService.reportExecutionError(
+      execution.command.toString(),
+      (execution.stderrSubject as any) as Observable<Error | undefined>
+    );
+    channelService.streamCommandOutput(execution);
+    ProgressNotification.show(execution, cancellationTokenSource);
+    taskViewService.addCommandExecution(execution, cancellationTokenSource);
+  }
+}
+
+const workspaceChecker = new SfdxWorkspaceChecker();
+const parameterGatherer = new EmptyParametersGatherer();
+
+export async function forceDescribeMetadata(outputPath?: string) {
+  if (isNullOrUndefined(outputPath)) {
+    outputPath = await getMetadataTypesPath();
+  }
+  const describeExecutor = new ForceDescribeMetadataExecutor(outputPath!);
+  const commandlet = new SfdxCommandlet(
+    workspaceChecker,
+    parameterGatherer,
+    describeExecutor
+  );
+  await commandlet.run();
+}
+
+export async function getMetadataTypesPath(): Promise<string | undefined> {
+  if (hasRootWorkspace()) {
+    const workspaceRootPath = getRootWorkspacePath();
+    const defaultUsernameOrAlias = await OrgAuthInfo.getDefaultUsernameOrAlias();
+    const defaultUsernameIsSet = typeof defaultUsernameOrAlias !== 'undefined';
+
+    if (defaultUsernameIsSet) {
+      const username = await OrgAuthInfo.getUsername(defaultUsernameOrAlias!);
+      const metadataTypesPath = path.join(
+        workspaceRootPath,
+        '.sfdx',
+        'orgs',
+        username,
+        'metadata',
+        'metadataTypes.json'
+      );
+      return metadataTypesPath;
+    } else {
+      const err = nls.localize('error_no_default_username');
+      telemetryService.sendError(err);
+      throw new Error(err);
+    }
+  } else {
+    const err = nls.localize('cannot_determine_workspace');
+    telemetryService.sendError(err);
+    throw new Error(err);
+  }
+}
+
+export type MetadataObject = {
+  directoryName: string;
+  inFolder: boolean;
+  metaFile: boolean;
+  suffix: string;
+  xmlName: string;
+};
+
+export function buildTypesList(metadataTypesPath: string): string[] {
+  if (fs.existsSync(metadataTypesPath)) {
+    const fileData = JSON.parse(fs.readFileSync(metadataTypesPath, 'utf8'));
+    const metadataObjects = fileData.metadataObjects as MetadataObject[];
+    const metadataTypes = [];
+    for (const metadataObject of metadataObjects) {
+      if (!isNullOrUndefined(metadataObject.xmlName)) {
+        metadataTypes.push(metadataObject.xmlName);
+      }
+    }
+    telemetryService.sendMetadataTypes(undefined, {
+      metadataTypes: metadataTypes.length
+    });
+    return metadataTypes;
+  } else {
+    const err =
+      'There was an error retrieving metadata type information. Refresh the view to retry.';
```

Comment:
@ruthemmanuelle If we run into an error when trying to retrieve the metadata types from the generated output file, I think we should throw an error. If the user runs into this scenario, they should be able to refresh the view to generate the output file and load the view again.

Reply:
Sorry, I just saw this.  Can we tell them how to refresh the view? Or is it obvious? Not sure what this UI looks like.

Reply:
I've taken out this message with my latest change, but at some point we will have to display a message similar to this. I think the view will include a refresh button in the top right corner, so we would want to point the user to that.

Reply:
Ok, it sounds like refreshing will be intuitive, so the message you had (or something similar to it) should be good. Let me know when you're ready to add the new message.